balancer/pickfirst/pickfirstleaf/pickfirstleaf_ext_test.go

```diff
@@ -58,7 +58,13 @@ const (
 	stateStoringBalancerName = "state_storing"
 )
 
-var stateStoringServiceConfig = fmt.Sprintf(`{"loadBalancingConfig": [{"%s":{}}]}`, stateStoringBalancerName)
+var (
+	stateStoringServiceConfig = fmt.Sprintf(`{"loadBalancingConfig": [{"%s":{}}]}`, stateStoringBalancerName)
+	ignoreBalAttributesOpt    = cmp.Transformer("IgnoreBalancerAttributes", func(a resolver.Address) resolver.Address {
```

Comment:
Should we have a test to ensure that the `managedByPickfirstKeyType` attribute is set on all calls from PR to create a new subchannel?

Reply:
I can add a test for this. Instead of testing for the balancer attribute directly, I added an outlier detection test that verifies that OD doesn't eject subchannels created by pickfirst when functioning as a top-level policy. On seeing the balancer attribute, OD will wait wait for pickfirst to register a health listener. Since a top-level pickfirst  doesn't register a health listener, it's subchannels can't be ejected.